internal/restic/snapshot_policy.go

```diff
@@ -288,6 +298,17 @@ func ApplyPolicy(list Snapshots, p ExpirePolicy) (keep, remove Snapshots, reason
 			}
 		}
 
+		if p.Unique {
+			if cur.Parent != nil {
+				parent := findParentSnapshot(keep, *cur.Parent)
+				if parent != nil {
+					if parent.Tree == cur.Tree {
+						keepSnap = false
+					}
+				}
+			}
+		}
```

Comment:
Not sure, if I have to do it before the keep and keep within retention passes or after. 

The more I think about it, I feel it should be before, but I don't know how that would interact with the other passes. 

I am open for thoughts.